gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineTesting.java

```diff
@@ -112,20 +120,32 @@ public void apply(Project project) {
         return Optional.empty();
     }
 
-    private static boolean hasCompileDependenciesMatching(Project project, SourceSet sourceSet, Spec<Dependency> spec) {
-        return project
-                .getConfigurations()
-                .getByName(sourceSet.getCompileClasspathConfigurationName())
-                .getAllDependencies()
-                .matching(spec)
-                .stream()
-                .findAny()
-                .isPresent();
+    private static void ifHasResolvedCompileDependenciesMatching(
+            Project project, SourceSet sourceSet, Predicate<ModuleComponentIdentifier> spec, Runnable runnable) {
+        project.getConfigurations()
+                .getByName(sourceSet.getRuntimeClasspathConfigurationName())
+                .getIncoming()
+                .afterResolve(deps -> {
```

Comment:
Since we need to distinguish between Spock 1 and Spock 2, we need to actually resolve the configuration to get a version number. It's actually quite tricky to resolve here, as it fails foul of the checks we have to prevent [resolving configuration at configuration time](https://docs.gradle.org/current/userguide/performance.html#dont_resolve_dependencies_at_configuration_time), which destroys perf.

Instead, we wait for this configuration to resolved, which will happen before the test is run, by the `checkJUnitDependencies` task if nothing else...

(This would be so much easier if `useJUnitPlatform` was a lazy property)

Reply:
I was initially thinking that it might be sufficient to just go through the explicitly declared deps which would not require resolution. However most of the time Spock is brought in as a transitive and so you have to do resolution :(

Reply:
I initially did that and checked for nebula-test too, but until resolution happens you don't know the version number, so can't decide whether or not to do the right thing